syntax.Rmd

```diff
@@ -438,7 +438,7 @@ switch(x, a =, b = 1, c = 2)
 switch(y, 1, 2, 3)
 ```
 
-## Long lines
+## Long function calls
 
 Strive to limit your code to 80 characters per line. This fits comfortably on a
 printed page with a reasonably sized font. If you find yourself running out of
```

Comment:
> Strive to limit your code to 80 characters per line. This fits comfortably on a
printed page with a reasonably sized font. If you find yourself running out of
room, this is a good indication that you should encapsulate some of the work in
a separate function.

Just noting that this advice feels kind of odd. We are talking about _wide_ lines here. I'm not sure that a wide line means you need a separate function - it probably just means you need to reflow the current one somehow?

Reply:
IME it's not _totally_ unrelated, the first thing that comes to mind is really long conditions in `if`:

```r
if (
  my_outer_call(my_inner_call(x)[[2]], other_arguments, one_keyword_arg = another_variable) &&
  ...
)
```

The suggestion here is to put that whole `my_outer_call(...)` as a helper. Yes, it can be re-flowed, but it increases nesting & doing so inside `if` is always a mess.

Reply:
We were discussing multiline if conditions a little bit internally too, I often create an informatively named boolean when I hit cases kind of like that

```r
is_an_outer_call <- my_outer_call(
  my_inner_call(x)[[2]], 
  other_arguments, 
  one_keyword_arg = another_variable
)

if (is_an_outer_call && is_this_other_thing)
```